tutorials/tour/_posts/2017-02-13-self-types.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: tutorial
+title: Self-type
+
+disqus: true
+
+tutorial: scala-tour
+categories: tour
+num: 25
+next-page: implicit-parameters
+previous-page: compound-types
+prerequisite-knowledge: nested-classes, mixin-class-composition
+---
+When writing extensible code, it can be useful to declare a self-type as a means of dependency injection.
```

Comment:
Not to minimize the difficulty, but this is quite opaque. Maybe something like: `Self-types are a way to declare that a trait must be mixed into another trait, even though it doesn't directly extend it. That makes the members of the dependency available without imports.`